tests/sources/test_dropbox_aiohttp.py

```diff
@@ -0,0 +1,758 @@
+
+    with patch.object(
+        aiohttp.ClientSession,
+        "post",
+        return_value=get_json_mock(mock_response=MOCK_ACCESS_TOKEN, status=200),
+    ):
+        await source.dropbox_client._set_access_token()
+        assert source.dropbox_client.access_token == "test2344"
+
+
+@pytest.mark.asyncio
+@mock.patch("connectors.utils.apply_retry_strategy")
+async def test_set_access_token_with_incorrect_app_key_then_raise_exception(
+    mock_apply_retry_strategy,
+):
+    source = create_source(DropboxDataSource)
+    mock_apply_retry_strategy.return_value = mock.Mock()
+
+    with patch.object(
+        aiohttp.ClientSession,
+        "post",
+        return_value=get_json_mock(
+            mock_response=MOCK_ACCESS_TOKEN_FOR_INVALID_APP_KEY, status=400
+        ),
+    ):
+        with pytest.raises(
+            InvalidClientCredentialException,
+            match="Configured App Key or App Secret is invalid.",
+        ):
+            await source.dropbox_client._set_access_token()
+
+
+@pytest.mark.asyncio
+@mock.patch("connectors.utils.apply_retry_strategy")
+async def test_set_access_token_with_incorrect_refresh_token_then_raise_exception(
+    mock_apply_retry_strategy,
+):
+    source = create_source(DropboxDataSource)
+    mock_apply_retry_strategy.return_value = mock.Mock()
+
+    with patch.object(
+        aiohttp.ClientSession,
+        "post",
+        return_value=get_json_mock(
+            mock_response=MOCK_ACCESS_TOKEN_FOR_INVALID_REFRESH_TOKEN, status=400
+        ),
+    ):
+        with pytest.raises(
+            InvalidRefreshTokenException, match="Configured Refresh Token is invalid."
+        ):
+            await source.dropbox_client._set_access_token()
+
+
+def test_tweak_bulk_options():
+    source = create_source(DropboxDataSource)
+
+    source.concurrent_downloads = 10
+    options = {"concurrent_downloads": 5}
+
+    source.tweak_bulk_options(options)
+    assert options["concurrent_downloads"] == 10
+
+
+@pytest.mark.asyncio
+async def test_close_with_client_session():
+    source = create_source(DropboxDataSource)
+    _ = source.dropbox_client._get_session
+
+    await source.close()
+    assert hasattr(source.dropbox_client.__dict__, "_get_session") is False
+
+
+@pytest.mark.asyncio
+async def test_ping():
+    source = create_source(DropboxDataSource)
+    source.dropbox_client._set_access_token = AsyncMock()
+    with patch.object(
+        aiohttp.ClientSession,
+        "post",
+        return_value=get_json_mock(MOCK_CURRENT_USER, 200),
+    ):
+        await source.ping()
+
+
+@pytest.mark.asyncio
+@patch("aiohttp.ClientSession.post")
+async def test_ping_with_ssl(
+    mock_post,
+):
+    mock_post.return_value.__aenter__.return_value.status = 200
+    source = create_source(DropboxDataSource)
+
+    source.dropbox_client.ssl_enabled = True
+    source.dropbox_client.certificate = (
+        "-----BEGIN CERTIFICATE----- Certificate -----END CERTIFICATE-----"
+    )
```

Comment:
custom client certs and enabling/disabling SSL isn't part of the connector configuration. Does it need to be?